### PR TITLE
Use Builder Pattern for HsmKeyParams

### PR DESCRIPTION
**Pull request by cps-b**
Adapt the HsmKeyParams to use a Builder Pattern instead of a struct

### CODE REVIEW

File: src/mococrw/hsm.h [hunk 1]
```diff
@@ -26,17 +26,49 @@ class ECCSpec;
...
+ * This class currently contains PKCS#11 attributes which are changeable on key creation.
  * In the future also parameters for other keystorage interfaces can be added.
  */
-struct HsmKeyParams
+class HsmKeyParams
 {
+public:
+    class Builder;
+
+    bool isExtractable() const { return cka_extractable; }
```
Comment (garw) [inline src/mococrw/hsm.h]: Do we really want to expose PKCS#11 slang here? Wouldn't it be better if this provides some Abstraction and we say here that the param is that the is is extractable or not? Sensitive is really not self-explanatory.

File: src/hsm.cpp [hunk 2]
```diff
@@ -233,7 +234,8 @@ openssl::SSL_EVP_PKEY_Ptr HsmEngine::generateKey(const ECCSpec &spec,
                                                  const std::string &keyLabel,
                                                  const std::vector<uint8_t> &keyID)
 {
-    HsmKeyParams hsmKeyParams;
+    HsmKeyParams hsmKeyParams =
+            HsmKeyParams::Builder().setCkaExtractable(false).setCkaSensitive(true).build();
```
Comment (garw) [inline src/hsm.cpp]: Style: Most of the library uses {} for object construction.

**Review by garw — CHANGES_REQUESTED** [review]

File: src/mococrw/hsm.h [hunk 3]
```diff
@@ -34,34 +34,25 @@ class HsmKeyParams
...
 
-    bool isExtractable() const { return cka_extractable; }
-
-    bool isSensitive() const { return cka_sensitive; }
+    bool isExtractable() const { return extractable; }
 
 private:
-    bool cka_extractable;
-    bool cka_sensitive;
+    bool extractable;
```
Comment (garw) [inline src/mococrw/hsm.h]: consistency: this lib prefixes private members with _

**Review by garw — DISMISSED** [review]
looks cleaner to me now. Feel free to merge after the consistency problem has been fixed.

**Comment by cps-b** [comment]
Merging in PR #156

**Pull request closed without merging**